File: external/cec2006_src/cec2006-benchmarks-master/code/matlab/g06.m
function f = g06_objective(x)
% CEC 2006 g06 constrained optimization problem objective function.
%
% Parameters:
% x: Decision variables [x1, x2]
%
% Returns:
% f: Objective function value

f = (x(1) - 10)^3 + (x(2) - 20)^3;

function g = g06_constraints(x)
% CEC 2006 g06 constraints.
%
% Parameters:
% x: Decision variables [x1, x2]
%
% Returns:
% g: Constraint values (should be <= 0 for feasible)

g = zeros(2, 1);
g(1) = -(x(1) - 5)^2 - (x(2) - 5)^2 + 100;
g(2) = (x(1) - 6)^2 + (x(2) - 5)^2 - 82.81;

function bounds = g06_bounds()
% Returns the bounds for g06 problem.
%
% Returns:
% bounds: [lower, upper] for each variable

bounds = [13, 100; 0, 100];

% Example usage
if ~exist('x', 'var')
    % Example solution (best known)
    x_opt = [14.095; 0.84296];
    f_opt = g06_objective(x_opt);
    g_opt = g06_constraints(x_opt);
    
    fprintf('Objective value: %f\n', f_opt);
    fprintf('Constraints: ');
    disp(g_opt');
    fprintf('All constraints satisfied: %d\n', all(g_opt <= 0));
end

File: external/cec2006_src/cec2006-benchmarks-master/code/python/g06.py
import numpy as np

def g06_objective(x):
    """
    CEC 2006 g06 constrained optimization problem objective function.
    
    Parameters:
    x (array-like): Decision variables [x1, x2]
    
    Returns:
    float: Objective function value
    """
    return (x[0] - 10)**3 + (x[1] - 20)**3

def g06_constraints(x):
    """
    CEC 2006 g06 constraints.
    
    Parameters:
    x (array-like): Decision variables [x1, x2]
    
    Returns:
    list: Constraint values (should be <= 0 for feasible)
    """
    g = []
    g.append(-(x[0] - 5)**2 - (x[1] - 5)**2 + 100)
    g.append((x[0] - 6)**2 + (x[1] - 5)**2 - 82.81)
    return g

def g06_bounds():
    """
    Returns the bounds for g06 problem.
    
    Returns:
    list: [(lower, upper), ...] for each variable
    """
    return [(13, 100), (0, 100)]

# Example usage
if __name__ == "__main__":
    # Example solution (best known)
    x_opt = [14.095, 0.84296]  # Approximate
    f_opt = g06_objective(x_opt)
    g_opt = g06_constraints(x_opt)
    
    print(f"Objective value: {f_opt}")
    print(f"Constraints: {g_opt}")
    print(f"All constraints satisfied: {all(g <= 0 for g in g_opt)}")